### PR TITLE
feat(cli): xcm transfer

### PR DESCRIPTION
**Pull request by 3h4x**
### Summary by OpenAI

**New Feature**
- Added a new option `-x, --xtransfer [file-path]` to the `init` command in the CLI program for generating a cross-chain transfer template.
- Introduced a new side effect action "XcmTransfer" with configuration values retrieved from environment variables or default fallbacks.
- Enhanced the `Gateway` class with a new method `createXTransferSfx` for creating a cross-chain transfer side effect.
- Added a new utility function `encodeXTransferArgs` for encoding arguments for cross-chain transfers.
- Integrated functions for generating an XCM reserve asset transfer message in a Polkadot application.

> 🚀 Across chains we now can glide,  
> With XCM transfers by our side.  
> Code that dances, code that sings,  
> In this release, cross-chain brings! 🎉

### CODE REVIEW

**Review by 3h4x — COMMENTED** [review]
I created this PR so I can't approve.

It looks good for me I just wonder why you haven't used `generateXcmReserveAssetTransferMessageParameters` and others?

Also if we use `@polkadot/types/interfaces` why won't you pin it's version?

File: Cargo.toml [hunk 1]
```diff
@@ -200,6 +200,9 @@ substrate-prometheus-endpoint  = { git = "https://github.com/paritytech/substrat
 substrate-wasm-builder         = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
 try-runtime-cli                = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
 
+# Temproary Asset Registry
```
Comment (3h4x) [inline Cargo.toml:203]: Resolve conflict

File: client/packages/sdk/package.json [hunk 2]
```diff
@@ -40,6 +40,7 @@
   "dependencies": {
     "@polkadot/api": "^8.9.1",
     "dotenv": "^16.0.3",
+    "interfaces": "link:@polkadot/types/interfaces",
```
Comment (3h4x) [inline client/packages/sdk/package.json:43]: Why link?

File: Cargo.toml (hunk 1 above)
Comment (chexware) [reply Cargo.toml:203]: to be removed. New AssetRegistry already on development branch

File: client/packages/sdk/package.json (hunk 2 above)
Comment (chexware) [reply client/packages/sdk/package.json:43]: Was created by default when installing the dependencies

Comment (3h4x) [reply client/packages/sdk/package.json:43]: Pin it to a version

**Comment by 3h4x** [comment]
Change in SDK require it version to updated so package can be published on NPM

File: client/packages/sdk/package.json (hunk 2 above)
Comment (chexware) [reply client/packages/sdk/package.json]: just updated it

**Review by MaciejBaj — APPROVED** [review]

**Comment by chexware** [comment]
> Change in SDK require it version to updated so package can be published on NPM

Is this the version that needs to be updated?
```
program
  .name("t3rn CLI")
  .description("CLI for interacting with the t3rn blockchain")
  .version("0.1.1")
```

File: client/packages/cli/src/cli.ts [hunk 3]
```diff
@@ -19,12 +19,13 @@ const program = new Command()
 program
   .name("t3rn CLI")
   .description("CLI for interacting with the t3rn blockchain")
-  .version("0.1.0")
+  .version("0.1.1")
```
Comment (chexware) [inline client/packages/cli/src/cli.ts:22]: Is this the version that needs to be updated @3h4x

Comment (3h4x) [reply client/packages/cli/src/cli.ts:22]: Yeah but 0.1.1 is already released so it will fail.

Comment (3h4x) [reply client/packages/cli/src/cli.ts:22]: @chexware packages/sdk/package.json version

**Comment by chexware** [comment]
@3h4x Updated the CLI to generate the parameters for an XCM transfer using the `pnpm cli xcmTransfer` command. Let me if I need to add anything else on that end before updating the command to submit an XCM transaction.

**Pull request merged**